Sources/Testing/Running/Configuration.TestFilter.swift

```diff
@@ -0,0 +1,295 @@
+      ///   - lhs: The first test filter's kind.
+      ///   - rhs: The second test filter's kind.
+      ///
+      /// The result of a test filter with this kind is the combination of the
+      /// results of its subfilters (as if with `&&`.)
+      indirect case combined(_ lhs: Kind, _ rhs: Kind)
+    }
+
+    /// The kind of test filter.
+    private var _kind: Kind
+
+    /// Whether or not to include tests with the `.hidden` trait when filtering
+    /// tests.
+    ///
+    /// By default, any test with the `.hidden` trait is treated as if it did
+    /// not pass the test filter's predicate function. When the testing library
+    /// is running its own tests, it sometimes uses this property to enable
+    /// discovery of fixture tests.
+    ///
+    /// The value of this property is inherited from `self` when using
+    /// ``combining(with:)`` or ``combine(with:)`` (i.e. the left-hand test
+    /// filter takes precedence.)
+    ///
+    /// This property is not part of the public interface of the testing
+    /// library.
+    var includeHiddenTests = false
+  }
+}
+
+// MARK: - Initializers
+
+extension Configuration.TestFilter {
+  /// A test filter that does not perform any filtering.
+  ///
+  /// This test filter allows all tests to run; it is the default test filter if
+  /// another is not specified.
+  public static var unfiltered: Self {
+    Self(_kind: .unfiltered)
+  }
+
+  /// Initialize this instance to filter tests to those specified by a set of
+  /// test IDs.
+  ///
+  /// - Parameters:
+  ///   - testIDs: A set of test IDs to be filtered.
+  public init(including testIDs: some Collection<Test.ID>) {
+    let selection = Test.ID.Selection(testIDs: testIDs)
+    self.init(_kind: .precomputed(selection, membership: .including))
+  }
+
+  /// Initialize this instance to filter tests to those _not_ specified by a set
+  /// of test IDs.
+  ///
+  /// - Parameters:
+  ///   - selection: A set of test IDs to be excluded.
+  public init(excluding testIDs: some Collection<Test.ID>) {
+    let selection = Test.ID.Selection(testIDs: testIDs)
+    self.init(_kind: .precomputed(selection, membership: .excluding))
+  }
+
+  /// Initialize this instance from an arbitrary function.
+  ///
+  /// - Parameters:
+  ///   - membership: How to interpret the result when predicating tests.
+  ///   - predicate: The function to predicate tests against.
+  init(membership: Membership, matching predicate: @escaping @Sendable (borrowing Test) -> Bool) {
+    self.init(_kind: .function(predicate, membership: membership))
+  }
+
+  /// Initialize this instance to operate based on a set of tags.
+  ///
+  /// - Parameters:
+  ///   - tags: The set of tags to either include or exclude.
+  ///   - anyOf: Whether tests must have _any_ of the tags in `tags` (as opposed
+  ///     to all of them.)
+  ///   - membership: How to interpret the result when predicating tests.
+  init(tags: some Collection<Tag>, anyOf: Bool, membership: Membership) {
```

Comment:
I feel like it might be worth exposing this `init`, which accepts the `anyOf:` and `membership:` parameters, directly. Otherwise, a tool adopting this would need to selectively call one of the more specific `init` overloads below using logic that would seem cleaner if they could just pass these values directly.

I realize, as it's currently written this suggestion would require elevating the `Membership` enum to `public`, but It seems like instead of doing that, this parameter could be represented as a plain `Bool` type? I.e.

```swift
public init(tags: some Collection<Tag>, anyOf: Bool, including: Bool) {
  // ...convert `including` to `Membership` first, then call the existing init...
}
```

Reply:
I intentionally did not include such an overload because it raises the aesthetic question of whether `including` or `excluding` should be `true` or `false`, and how to properly represent `anyOf` vs. `allOf` (presumably another enum.)

Since this is a tools-only interface, we can always improve upon it later!